src/impl_dyn.rs

```diff
@@ -52,10 +52,62 @@ where S: Data<Elem = A>
     /// assert_eq!(a.shape(), &[2]);
     /// ```
     #[track_caller]
-    pub fn index_axis_inplace(&mut self, axis: Axis, index: usize)
-    {
+    pub fn index_axis_inplace(&mut self, axis: Axis, index: usize) {
         self.collapse_axis(axis, index);
         self.dim = self.dim.remove_axis(axis);
         self.strides = self.strides.remove_axis(axis);
     }
+
+    /// Remove axes of length 1 and return the modified array.
+    ///
+    /// If the array has more the one dimension, the result array will always
+    /// have at least one dimension, even if it has a length of 1.
+    ///
+    /// ```
+    /// use ndarray::{arr2, arr3};
+    ///
+    /// let a = arr3(&[[[1, 2, 3]], [[4, 5, 6]]]).into_dyn();
+    /// assert_eq!(a.shape(), &[2, 1, 3]);
+    /// let b = a.squeeze();
+    /// assert_eq!(b, arr2(&[[1, 2, 3], [4, 5, 6]]).into_dyn());
+    /// assert_eq!(b.shape(), &[2, 3]);
+    ///
+    /// let c = arr2(&[[1]]).into_dyn();
+    /// assert_eq!(c.shape(), &[1, 1]);
+    /// let d = c.squeeze();
+    /// assert_eq!(d, arr1(&[1]).into_dyn());
```

Comment:
You use `arr1` but you do not import it.

Reply:
right, fixed